api/envoy/config/rbac/v3/rbac.proto

```diff
@@ -218,6 +219,10 @@ message Permission {
     // Please refer to :ref:`this FAQ entry <faq_how_to_setup_sni>` to learn to
     // setup SNI.
     type.matcher.v3.StringMatcher requested_server_name = 9;
+
+    // Optional extension for configuring custom matchers for RBAC.
+    // [#extension-category: envoy.filters.common.rbac.matcher]
+    core.v3.TypedExtensionConfig matcher = 12;
```

Comment:
CC @wjtracey who is looking into CEL attribute extensibility. This is different, since its fixed matcher extension, but there is some overlap.